Mathlib/Algebra/Group/TypeTags.lean

```diff
@@ -0,0 +1,465 @@
+
+## See also
+
+This file is similar to `Order.Synonym`.
+
+## Porting notes
+
+- Since bundled morphism applications that rely on `CoeFun` currently don't work, they are ported
+  as `toFoo a` rather than `a.toFoo` for now. (https://github.com/leanprover/lean4/issues/1910)
+
+-/
+
+universe u v
+
+variable {α : Type u} {β : Type v}
+
+/-- If `α` carries some multiplicative structure, then `Additive α` carries the corresponding
+additive structure. -/
+def Additive (α : Type _) := α
+#align additive Additive
+
+/-- If `α` carries some additive structure, then `Multiplicative α` carries the corresponding
+multiplicative structure. -/
+def Multiplicative (α : Type _) := α
+#align multiplicative Multiplicative
+
+namespace Additive
+
+/-- Reinterpret `x : α` as an element of `additive α`. -/
+def ofMul : α ≃ Additive α :=
+  ⟨fun x => x, fun x => x, fun _ => rfl, fun _ => rfl⟩
+#align additive.of_mul Additive.ofMul
+
+/-- Reinterpret `x : additive α` as an element of `α`. -/
+def toMul : Additive α ≃ α := ofMul.symm
+#align additive.to_mul Additive.toMul
+
+@[simp]
+theorem ofMul_symm_eq : (@ofMul α).symm = toMul :=
+  rfl
+#align additive.of_mul_symm_eq Additive.ofMul_symm_eq
+
+@[simp]
+theorem toMul_symm_eq : (@toMul α).symm = ofMul :=
+  rfl
+#align additive.to_mul_symm_eq Additive.toMul_symm_eq
+
+end Additive
+
+namespace Multiplicative
+
+/-- Reinterpret `x : α` as an element of `multiplicative α`. -/
+def ofAdd : α ≃ Multiplicative α :=
+  ⟨fun x => x, fun x => x, fun _ => rfl, fun _ => rfl⟩
+#align multiplicative.of_add Multiplicative.ofAdd
+
+/-- Reinterpret `x : multiplicative α` as an element of `α`. -/
+def toAdd : Multiplicative α ≃ α := ofAdd.symm
+#align multiplicative.to_add Multiplicative.toAdd
+
+@[simp]
+theorem ofAdd_symm_eq : (@ofAdd α).symm = toAdd :=
+  rfl
+#align multiplicative.of_add_symm_eq Multiplicative.ofAdd_symm_eq
+
+@[simp]
+theorem toAdd_symm_eq : (@toAdd α).symm = ofAdd :=
+  rfl
+#align multiplicative.to_add_symm_eq Multiplicative.toAdd_symm_eq
+
+end Multiplicative
+
+open Additive (ofMul toMul)
+open Multiplicative (ofAdd toAdd)
+
+@[simp]
+theorem toAdd_ofAdd (x : α) : toAdd (ofAdd x) = x :=
+  rfl
+#align to_add_of_add toAdd_ofAdd
+
+@[simp]
+theorem ofAdd_toAdd (x : Multiplicative α) : ofAdd (toAdd x) = x :=
+  rfl
+#align of_add_to_add ofAdd_toAdd
+
+@[simp]
+theorem toMul_ofMul (x : α) : toMul (ofMul x) = x :=
+  rfl
+#align to_mul_of_mul toMul_ofMul
+
+@[simp]
+theorem ofMul_toMul (x : Additive α) : ofMul (toMul x) = x :=
+  rfl
+#align of_mul_to_mul ofMul_toMul
+
+instance [Inhabited α] : Inhabited (Additive α) :=
+  ⟨ofMul default⟩
+
+instance [Inhabited α] : Inhabited (Multiplicative α) :=
+  ⟨ofAdd default⟩
+
+instance [Finite α] : Finite (Additive α) :=
+  Finite.of_equiv α (by rfl)
+
+instance [Finite α] : Finite (Multiplicative α) :=
+  Finite.of_equiv α (by rfl)
+
+instance [h: Infinite α] : Infinite (Additive α) := h
```

Comment:
This used to be `by tauto` but it is not (?) ported yet.

Reply:
Please leave porting notes in cases like this, so we have a change to restore the tactic proofs later.